src/test/scala/ai/privado/languageEngine/java/AbstractTaggingSpec.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * This file is part of Privado OSS.
+ *
+ * Privado is an open source static code analysis tool to discover data flows in the code.
+ * Copyright (C) 2022 Privado, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * For more information, contact support@privado.ai
+ *
+ */
+
+package ai.privado.languageEngine.java
+
+import ai.privado.cache.{AppCache, RuleCache}
+import ai.privado.model.*
+import ai.privado.model.Language.{JAVA, KOTLIN, Language}
+import better.files.File
+import io.joern.javasrc2cpg.JavaSrc2Cpg
+import io.joern.kotlin2cpg.Kotlin2Cpg
+import io.joern.x2cpg.X2Cpg.applyDefaultOverlays
+import io.shiftleft.codepropertygraph.generated.Cpg
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach}
+
+abstract class AbstractTaggingSpec(val language: Language)
```

Comment:
New abstract class that makes each test isolated, with no global variables and takes  a language as a parameter (only Kotlin and Java CPG branches right now but others can be added easily). I am not using this base class everywhere, only in the new tests.